lib/view.js

```diff
@@ -99,10 +88,11 @@ function view (args, silent, cb) {
 
   if (local) {
     const dir = npm.prefix
-    BB.resolve(readJson(path.resolve(dir, 'package.json'))).nodeify((er, d) => {
-      d = d || {}
-      if (er && er.code !== 'ENOENT' && er.code !== 'ENOTDIR') return cb(er)
-      if (!d.name) return cb(new Error('Invalid package.json'))
+    readJson(path.resolve(dir, 'package.json')).catch(er => {
+      if (er && er.code !== 'ENOENT' && er.code !== 'ENOTDIR') throw er
+      return {}
+    }).then(d => {
+      if (!d || !d.name) return cb(new Error('Invalid package.json'))
```

Comment:
This was me just mechanically keeping the same logic from before, so I can't really defend the choice here.  But, as I read this, that code path would get hit if some other error happened.  (Eg, you don't have read permission to the package.json, and get an `EACCES` error, or `path/package.json/` is a directory, so you get `EISDIR`, etc.  Any error from [read](http://man7.org/linux/man-pages/man2/read.2.html#ERRORS) or [open](http://man7.org/linux/man-pages/man2/open.2.html#ERRORS), really.)

The confusing part is that in those cases, we fail with a generic `Invalid package.json`, rather than the actual ENOENT/ENOTDIR error encountered reading the file, which seems like it can only obscure things.  I'm honestly not sure why it's there in the first place.  If I had to _guess_, I'd think that it's likely moved or copied from a place where it was once upon a time handling a missing package in some reasonable way (but letting other errors go ahead and fail).

I think we should probably remove this catch block.

Reply:
Zooming out a bit on this, this is only relevant if the local prefix doesn't have a package.json in it.  So the impact here is that we get this:

```
$ npm view
npm ERR! Invalid package.json

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/isaacs/.npm/_logs/2020-03-05T06_38_26_462Z-debug.log
```

instead of this

```
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /Users/isaacs/dev/npm/json-parse-even-better-errors/test/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/Users/isaacs/dev/npm/json-parse-even-better-errors/test/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/isaacs/.npm/_logs/2020-03-05T06_37_52_426Z-debug.log
```

And I'm inclined to think that the proper `ENOENT` is more helpful, since it includes the path and the "this is about not being able to find a file" pointer.

Removing.